util/cache.go

```diff
@@ -0,0 +1,111 @@
+// Handle the password caching
+package util
+
+import (
+	"errors"
+	"time"
+
+	"github.com/patrickmn/go-cache"
+)
+
+type CachedUser struct {
```

Comment:
Please comment all the exported items in this class, as per the go styling standard (gometalinter can help with finding these sort of style issues) https://blog.golang.org/godoc-documenting-go-code